script.js

```diff
@@ -1,100 +1,332 @@
+    document.getElementById("nombre-material").value = m.nombre;
+    document.getElementById("costo-material").value = m.costoUnitario;
+    document.getElementById("cantidad").value = m.cantidad;
+    document.getElementById("unidad-material").value = m.unidad;
+    document.getElementById("cantidad-equivalente").value = m.unidadesPorSeleccionada;
+
+    // Eliminar y esperar nuevo submit
+    calculadora.materiales.splice(index, 1);
+    guardarDatosEnStorage();
+    actualizarTablaMateriales();
+  }
+});
+
+
+// Evento para editar gasto fijo
+tablaGastosFijos.addEventListener("click", (e) => {
+// Prellena el formulario y elimina el gasto para reemplazarlo
+  if (e.target.dataset.tipo === "editar-gasto") {
+    const index = parseInt(e.target.dataset.index);
+    const g = calculadora.gastosFijos[index];
+
+    document.getElementById("concepto-fijo").value = g.concepto;
+    document.getElementById("costo-fijo").value = g.costo;
+
+    calculadora.gastosFijos.splice(index, 1);
+    guardarDatosEnStorage();
+    actualizarTablaGastosFijos();
+  }
+});
+
+tablaMateriales.addEventListener("click", (e) => {
+  // se confirma con el usuario antes de eliminar un material de la lista
+// se muestra botones de "Confirmar" y "Cancelar" dentro de la fila
+  if (e.target.dataset.tipo === "material") {
+    const fila = e.target.closest("tr");
+    const index = parseInt(e.target.dataset.index);
+
+    if (fila.querySelector(".btn-confirmar")) return;
+
+    fila.innerHTML = `
+      <td colspan="4" class="text-center text-danger">
+        ¿Seguro que quieres eliminar este material?
+        <button class="btn btn-danger btn-sm btn-confirmar" data-index="${index}">Confirmar</button>
+        <button class="btn btn-secondary btn-sm btn-cancelar">Cancelar</button>
+      </td>
+    `;
+
+    fila.querySelector(".btn-confirmar").addEventListener("click", () => {
+      calculadora.materiales.splice(index, 1);
+      guardarDatosEnStorage();
+      actualizarTablaMateriales();
+    });
+
+    fila.querySelector(".btn-cancelar").addEventListener("click", () => {
+      actualizarTablaMateriales();
+    });
+  }
+});
+
+tablaGastosFijos.addEventListener("click", (e) => {
+  // se confirma con el usuario antes de eliminar un gasto fijo
+// se muestra botones de "Confirmar" y "Cancelar" en la fila
+  if (e.target.dataset.tipo === "gasto") {
+    const fila = e.target.closest("tr");
+    const index = parseInt(e.target.dataset.index);
+
+    if (fila.querySelector(".btn-confirmar")) return;
+
+    fila.innerHTML = `
+      <td colspan="3" class="text-center text-danger">
+        ¿Seguro que quieres eliminar este gasto?
+        <button class="btn btn-danger btn-sm btn-confirmar" data-index="${index}">Confirmar</button>
+        <button class="btn btn-secondary btn-sm btn-cancelar">Cancelar</button>
+      </td>
+    `;
+
+    fila.querySelector(".btn-confirmar").addEventListener("click", () => {
+      calculadora.gastosFijos.splice(index, 1);
```

Comment:
Delega la responsabilidad a calculadora:
```suggestion
calculadora.eliminarGastoFijo(index) 
```